### PR TITLE
added local attendee Delete from event and sync scheduling after yhe …

### PR DESCRIPTION
**Pull request by viratde**
I have added support for deleting a user from an event and reschedling all the sync after first time logging in

### CODE REVIEW

File: agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstAgendaItemsRepository.kt [hunk 1]
```diff
@@ -116,4 +142,65 @@ class OfflineFirstAgendaItemsRepository(
...
+                syncType = SyncType.UPDATE,
+                userId = userId
+            ).forEach {
+                eventSyncScheduler.sync(EventSyncScheduler.SyncType.UpdateEventSync(it.event.toEvent()))
+            }
+            eventDeleteSyncDao.getAllEventDeletePendingSyncs(
+                userId = userId
+            ).forEach {
+                eventSyncScheduler.sync(EventSyncScheduler.SyncType.DeleteEventSync(it.eventId))
+            }
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstAgendaItemsRepository.kt:165]: In this function, the database gets accessed multiple times in a synchronous manner. So before you can schedule the reminder sync tasks, first the events and the task events need to complete. To me, that does not make much sense - why then have these operations depend on each other? 🤔 

In a situation like this, we can use Kotlin’s `async` coroutine builder to have all our operations happen asynchronously. Check out this sample:

```
override suspend fun syncPendingAgendaItems() {
  withContext(Dispatchers.IO) {
    val userId = authInfoStorage.get()?.userId ?: return@withContext
    
    val createdEvents = async {
      eventSyncDao.getAllEventPendingSyncs(
        syncType = SyncType.CREATE,
        userId = userId
      )
    }
    
    val deletedEvents = async {
      eventDeleteSyncDao.getAllEventDeletePendingSyncs(userId)
    }
    
    // .. other sync calls
    
    val createEventJobs = createdEvents.await().map { pendingEntity ->
      launch {
        val event = pendingEntity.event.toDomain() // convert to domain model Event
        
        // try to make the API call immediately
        when(remoteDataSource.postEvent(event)) {
          is Result.Error -> Unit
          is Result.Success -> applicationScope.launch {
            pendingSyncDao.deleteEventPendingSyncEntity(event.id)
          }.join()
        }
      }
    }
    
    // … other API calls
    
    // Then we await the completion of each before the function terminates
    createEventJobs.forEach { it.join() }
    deleteEventJobs.forEach { it.join() }
    createTaskJobs.forEach { it.join() }
    deletedTaskJobs.forEach { it.join() }
    // ...    
  }
}
```

File: agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstEventRepository.kt [hunk 2]
```diff
@@ -89,7 +91,22 @@ class OfflineFirstEventRepository(
...
+                eventSyncScheduler.sync(EventSyncScheduler.SyncType.UpdateEventSync(
+                    event.copy(
+                        attendees = event.attendees.filter {
+                            it.userId != userId || event.host == userId
+                        }
+                    )
+                ))
+            }.join()
+        }
+        localEventDataSource.deleteEvent(eventId)
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstEventRepository.kt:108]: If you’ve successfully performed the sync call, then we want to guarantee that the local is also deleted. `applicationScope` can be used to safeguard against premature cancellation of our coroutines.

Also, would this flow not make more sense for an offline-first app?
1. Delete the event app locally
2. If successful, attempt to delete the event remotely
3. If API call failed, then schedule a sync call

File: agenda/presentation/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -38,5 +38,5 @@
     <string name="edit">Edit</string>
     <string name="delete">Delete</string>
     <string name="logout">Logout</string>
-    <string name="photo_cannot_be_added">Photo can be added due to \n network inactivity.</string>
+    <string name="photo_cannot_be_added">Photo cann\'t be added.</string>
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/res/values/strings.xml:41]: minor - typo `can't`

**Review by Kai-Cilliers — APPROVED** [review]

File: agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstEventRepository.kt (hunk 2 above)
Comment (viratde) [reply agenda/data/src/main/java/com/tasky/agenda/data/repositories/OfflineFirstEventRepository.kt:108]: Yes

**Pull request merged**